client/src/controllers/RevealController.test.js

```diff
@@ -0,0 +1,617 @@
+import { Application } from '@hotwired/stimulus';
+
+import { RevealController } from './RevealController';
+
+jest.useFakeTimers();
+
+describe('RevealController', () => {
+  const eventNames = [
+    'w-reveal:closed',
+    'w-reveal:opened',
+    'w-reveal:ready',
+    'w-reveal:toggled',
+  ];
+
+  const events = {};
+
+  let application;
+  let errors = [];
+
+  beforeAll(() => {
+    eventNames.forEach((name) => {
+      events[name] = [];
+    });
+
+    Object.keys(events).forEach((name) => {
+      document.addEventListener(name, (event) => {
+        events[name].push(event);
+      });
+    });
+  });
+
+  const setup = async (
+    html = `
+    <section data-controller="w-reveal">
+      <button type="button" data-action="w-reveal#toggle" data-w-reveal-target="toggle" aria-controls="my-content">Toggle</button>
+      <div class="content" id="my-content">CONTENT</div>
+    </section>
+    `,
+    identifier = 'w-reveal',
+  ) => {
+    document.body.innerHTML = `<main>${html}</main>`;
+
+    application = new Application();
+
+    application.handleError = (error, message) => {
+      errors.push({ error, message });
+    };
+
+    application.register(identifier, RevealController);
+
+    application.start();
+
+    await jest.runAllTimersAsync();
+
+    return [
+      ...document.querySelectorAll(`[data-controller~="${identifier}"]`),
+    ].map((element) =>
+      application.getControllerForElementAndIdentifier(element, identifier),
+    );
+  };
+
+  afterEach(() => {
+    application?.stop && application.stop();
+    errors = [];
+    eventNames.forEach((name) => {
+      events[name] = [];
+    });
+  });
+
+  describe('when connecting to the DOM', () => {
+    it('should attach the aria-controls target from the toggle as the content target if not set', async () => {
+      const [controller] = await setup();
+      expect(controller.contentTarget).toEqual(
+        document.getElementById('my-content'),
+      );
+    });
+
+    it('should allow for a class to set the initial state of closed', async () => {
```

Comment:
Ah, yep, I just read the test after reviewing the implementation. This and the following two cases are what I meant, I think we should just consistently use the `closedValue`'s attribute as the single source of truth for the initial state, ignoring (and resolving) the others (class and aria-expanded) based on it.

Reply:
Yeah, I agree this is a bit convoluted, removing for now and IF we need it built this way for the panels implementation, we can revisit then.

Thanks for the feedback here.